marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs

```diff
@@ -503,3 +521,22 @@ failWhenFull x = do
   if isEmpty
   then pure ()
   else error "an indexer raised an error"
+
+drainUntil :: C.ChainPoint -> TChan (ChainSyncEvent (BlockInMode CardanoMode)) -> QSemN -> IO ()
+drainUntil cp tchan _barrier = go
+  where
+    go = do
+      signalQSemN _barrier 1
```

Comment:
Why do we need this here?

Reply:
It's to signal the coordinator to send new blocks, it stalls otherwise.